model-server/src/test/kotlin/org/modelix/model/server/handlers/ModelReplicationServerTest.kt

```diff
@@ -191,6 +192,15 @@ class ModelReplicationServerTest {
         }
     }
 
+    @Test
+    fun `responds with 404 when querying non-existent branch`() = runWithTestModelServer { _, _ ->
+        val repositoryId = "abc"
+        val branch = "non-existing-branch"
+        val response = client.post("/v2/repositories/$repositoryId/branches/$branch/query")
+        response shouldHaveStatus 404
+        assertContains(response.bodyAsText(), "Branch '$branch' does not exist in repository '$repositoryId'")
```

Comment:
Should we check whether the repository exists and give the user a better error message when the requested repository does not exist?

Reply:
Probably out-of-scope because we should do it consistently everywhere.